### ISSUE TITLE
Provide thumbnail with RSS

### ISSUE THREAD

**Issue by eskan** [LemmyNet/lemmy#581]
Hi,
the RSS feed should provide a url for image or thumbnail.
regards,

**Comment by dessalines** [comment]
I guess just the sites favicon? https://dev.lemmy.ml/static/assets/favicon.svg

**Comment by Nutomic** [comment]
Pretty sure the issue is about including thumbnails for posts in the RSS feed. Apparently we forgot to add that in the initial implementation.

**Comment by dessalines** [comment]
The thumbnails (unless its an image post) aren't stored on the back-end yet. This will have to wait for #558 .

**Comment by Nutomic** [comment]
We could also just include the full-size images.

**Comment by dessalines** [comment]
The pictshare_urls are full size, but the back end currently doesn't store any images, only the url field.

**Comment by eskan** [comment]
the url field should be perfect, here a rss reference : 
https://www.w3schools.com/xml/xml_rss.asp#rssref

**Comment by dessalines** [comment]
@Nutomic There are now thumbnails in the post fetch.

**Comment by Nutomic** [comment]
I might be missing something here, but it looks like the `rss` crate does not support images for feed items, only for channels. Maybe there is a workaround with extensions or something, but I'm not sure.

https://rust-syndication.github.io/rss/rss/struct.Channel.html#method.image
https://rust-syndication.github.io/rss/rss/struct.Item.html

**Comment by dessalines** [comment]
[Unfortunately stackoverflow is debating on it](https://stackoverflow.com/questions/705224/how-do-i-add-an-image-to-an-item-in-rss-2-0), but it looks like it needs to be an enclosure:

https://rust-syndication.github.io/rss/rss/struct.Enclosure.html

Basically if the pictshare_thumbnail column exists, we set an image enclosure for the item.

**Comment by Nutomic** [comment]
I dont really care, but anyway its really low priority.

**Comment by Nutomic** [comment]
This is easy enough to do with the following code, but [enclosure requires length and mime fields](https://cyber.harvard.edu/rss/rss.html#ltenclosuregtSubelementOfLtitemgt). We could get the mime based on the thumbnail filename, but getting the size requires a network call to pict-rs, and thats just too slow. 

```
    if let Some(thumbnail) = p.thumbnail_url {
      let mut enclosure = Enclosure::default();
      enclosure.set_url(thumbnail);
      i.enclosure(enclosure);
    }
```

**Comment by Elara6331** [comment]
I believe the `enclosure` tag is meant to be used for prefetching, and I don't think it gets shown as a thumbnail in most RSS feed readers. I tried it on a few and it didn't show up as a thumbnail.

What does seem to work in most readers is the `<media:content />` tag from the Media extension, which isn't supported by the `rss` crate yet (rust-syndication/rss#29). Specifically, this tag seems to work everywhere I've tried:

```rss
<media:content url="https://lemmy.ml/pictrs/image/84ccd4ee-a278-4bf0-9d63-dad21add2e50.jpeg?thumbnail=256" medium="image"/>
```

Here's what this looks like in Fluent Reader:

![](https://github.com/LemmyNet/lemmy/assets/41810442/51d95866-2be1-4e07-a15a-569c70cdc997)
